### DISCUSSION TITLE
为什么element-plus最外层package.json没有依赖@vue/shared，在packages/**中却能import

### DISCUSSION THREAD

**Discussion by apricity-site** [element-plus/element-plus#9371] [Q&A]
element-plus 的最外层package.json依赖没有@vue/shared，pnpm i 后node_module里却有@vue/shared。这是什么原理呢

**Comment by tang-tag** [comment]
在根目录的.npmrc中设置了shamefully-hoist=true，平铺了依赖，而vue依赖了@vue/shared,所以node_modules中也有 @vue/shared

**Reply by hanaviavia** [reply]
学习了，感谢
